packages/realm-network-transport/src/DefaultNetworkTransport.ts

```diff
@@ -31,6 +30,8 @@ import type {
 export class DefaultNetworkTransport implements NetworkTransport {
   public static fetch: Fetch;
   public static AbortController: AbortController;
+  // A hook to transform responses before being returned from fetch
+  public static transformResponse: (response: FetchResponse) => FetchResponse = (response) => response;
```

Comment:
I know we now use this hook for transforming the response, but since the hook doesn't necessarily have to transform anything we may want to call it something else? I'm thinking e.g. in the open issue where we've discussed exposing `DefaultNetworkTransport` to allow users to add their own `fetch`, they may also use this hook.

Don't have specific name suggestions atm, perhaps something with `middleware` (as it behaves similar to that)? But yeah would be interesting to hear if you think the original name would suffice in this case 👍

Reply:
The primary need for it to be a separate package, instead of being embedded in [the "platform" abstraction of the `realm` package](https://github.com/realm/realm-js/tree/main/packages/realm/src/platform), is because we use it for both `realm` and `realm-web`: If we move forward on plans to provide an SDK for atlas app services, without native modules & sync, we can delete the `@realm/network-transport` package all together.